File: www/js/classes/stream-case-parameter.js
class StreamCaseParameter {
    static fromJson(json) {
        if (json == null) {
            return null;
        }

        const parameterType = MostType.fromJson(json.parameterType);

        const streamCaseParameter = new StreamCaseParameter();

        streamCaseParameter.setId(json.id);
        streamCaseParameter.setParameterName(json.parameterName);
        streamCaseParameter.setParameterIndex(json.parameterIndex);
        streamCaseParameter.setParameterDescription(json.parameterDescription);
        streamCaseParameter.setParameterType(parameterType);

        return streamCaseParameter;
    }

    static toJson(streamCaseParameter) {
        if (streamCaseParameter == null) {
            return null;
        }
        return {
            id:                     streamCaseParameter.getId(),
            parameterName:          streamCaseParameter.getParameterName(),
            parameterIndex:         streamCaseParameter.getParameterIndex(),
            parameterDescription:   streamCaseParameter.getParameterDescription(),
            parameterType:          MostType.toJson(streamCaseParameter.getParameterType())
        };
    }

    constructor() {
        this._id                    = null;
        this._parameterName         = null;
        this._parameterIndex        = null;
        this._parameterDescription  = null;
        this._parameterType         = null;
    }

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setParameterName(parameterName) {
        this._parameterName = parameterName;
    }

    getParameterName() {
        return this._parameterName;
    }

    setParameterIndex(parameterIndex) {
        this._parameterIndex = parameterIndex;
    }

    getParameterIndex() {
        return this._parameterIndex;
    }

    setParameterDescription(parameterDescription) {
        this._parameterDescription = parameterDescription;
    }

    getParameterDescription() {
        return this._parameterDescription;
    }

    setParameterType(parameterType) {
        this._parameterType = parameterType;
    }

    getParameterType() {
        return this._parameterType;
    }

}

File: app/src/main/java/com/softwareverde/tidyduck/most/StreamCaseParameter.java
package com.softwareverde.tidyduck.most;

public class StreamCaseParameter {
    private Long _id;
    private String _parameterName;
    private String _parameterIndex;
    private String _parameterDescription;
    private MostType _parameterType;

    public Long getId() {
        return _id;
    }

    public void setId(Long id) {
        _id = id;
    }

    public String getParameterName() {
        return _parameterName;
    }

    public void setParameterName(String parameterName) {
        _parameterName = parameterName;
    }

    public String getParameterIndex() {
        return _parameterIndex;
    }

    public void setParameterIndex(String parameterIndex) {
        _parameterIndex = parameterIndex;
    }

    public String getParameterDescription() {
        return _parameterDescription;
    }

    public void setParameterDescription(String parameterDescription) {
        _parameterDescription = parameterDescription;
    }

    public MostType getParameterType() {
        return _parameterType;
    }

    public void setParameterType(MostType parameterType) {
        _parameterType = parameterType;
    }
}
